pandas/tests/indexes/datetimes/test_constructors.py

```diff
@@ -959,3 +959,74 @@ def test_pass_datetimeindex_to_index(self):
         expected = Index(rng.to_pydatetime(), dtype=object)
 
         tm.assert_numpy_array_equal(idx.values, expected.values)
+
+
+@pytest.mark.parametrize(
+    "ts_input,fold",
+    [
+        (1572136200000000000, 0),
+        ("2019-10-27 01:30:00+01:00", 0),
+        (datetime(2019, 10, 27, 0, 30, 0, 0, tzinfo=timezone.utc), 0),
+    ],
+)
+def test_timestamp_constructor_fold_conflict(ts_input, fold):
+    # Test for #25057
+    # Check that we raise on fold conflict
+    msg = (
+        "Cannot pass fold with possibly unambiguous input: int, float, "
+        "numpy.datetime64, str, or timezone-aware datetime-like. "
+        "Pass naive datetime-like or build Timestamp from components."
+    )
+    with pytest.raises(ValueError, match=msg):
+        Timestamp(ts_input=ts_input, fold=fold)
+
+
+@pytest.mark.parametrize("tz", ["dateutil/Europe/London", "Europe/London"])
```

Comment:
Maybe add `None` as well?

Reply:
Done. Another benefit of switching away from passing fold all over the place is that we automatically set fold, no matter if Timestamp is naive or not.